rootfs-builder/rootfs.sh

```diff
@@ -312,12 +332,39 @@ build_rootfs_distro()
 
 	echo "Required Go version: $GO_VERSION"
 
+	# need to detect rustc's version too?
+	detect_rust_version ||
+		die "Could not detect the required rust version for AGENT_VERSION='${AGENT_VERSION:-master}'."
+
+	echo "Required rust version: $RUST_VERSION"
+
+	detect_cmake_version ||
+		die "Could not detect the required cmake version for AGENT_VERSION='${AGENT_VERSION:-master}'."
+
+	echo "Required cmake version: $CMAKE_VERSION"
+
+	detect_musl_version ||
+		die "Could not detect the required musl version for AGENT_VERSION='${AGENT_VERSION:-master}'."
+
+	echo "Required musl version: $MUSL_VERSION"
+
 	if [ -z "${USE_DOCKER}" ] && [ -z "${USE_PODMAN}" ]; then
 		#Generate an error if the local Go version is too old
 		foundVersion=$(go version | sed -E "s/^.+([0-9]+\.[0-9]+\.[0-9]+).*$/\1/g")
 
-		compare_versions "$GO_VERSION" $foundVersion || \
-			die "Your Go version $foundVersion is older than the minimum expected Go version $GO_VERSION"
+		compare_versions "${GO_VERSION}" "${foundVersion}" || \
+			die "Your Go version ${foundVersion} is older than the minimum expected Go version ${GO_VERSION}"
+
+		if [ "${RUST_AGENT}" == "yes" ]; then
+			source /root/.cargo/env
```

Comment:
This doesn't look right to me - if I'm not using a container manager I would expect this script to use `$USER`'s rustup env, not the `root` users one? Under the CI, `$USER` may be `root`, but if a developer is running this, they don't want to have two instances of rustup (one for `$USER` and one for `root`) I think?